asmcli/tests/run_add_to_mesh

```diff
@@ -0,0 +1,151 @@
+#!/bin/bash
+set -CeEu
+set -o pipefail
+
+SPATH="$(readlink -f "$0")"
+SDIR="$(dirname "${SPATH}")"; export SDIR;
+SCRIPT_NAME="${0##*/}"
+
+# shellcheck source=common.sh
+. "${SDIR}/common.sh"
+
+cd "${SDIR}"
+
+get_test_clusters() {
+  cat <<EOF
+long-term-test-cluster-add-to-mesh-1
+long-term-test-cluster-add-to-mesh-2
+EOF
+}
+
+main() {
+  # CLI setup
+  parse_args "$@"
+  trap 'clean_up_clusters' ERR
+
+  set_up_clusters
+  run_add_to_mesh_test
+  clean_up_clusters
+}
+
+run_add_to_mesh_test() {
+  date +"%T"
+
+  local ARGS; ARGS="add-to-mesh ${PROJECT_ID}"
+  while read -r CLUSTER_NAME; do
+    ARGS="${ARGS} ${PROJECT_ID}/${CLUSTER_LOCATION}/${CLUSTER_NAME}"
+  done <<EOF
+$(get_test_clusters)
+EOF
+
+  # Test starts here
+  echo "Adding clusters to fleet..."
```

Comment:
If we're going to use long-term test clusters for this, we should validate that they're actually unregistered before we start the test, or we might get a false positive.

Reply:
I tried to unregister the clusters in setup and cleanup, will that suffice? If I check if it's unregistered before the test, and, say, it's registered, should we tried to unregister it (again), or fail the test (but it's our fault since we didn't properly set up the testing cluster)?

Reply:
Let's just fail. If we tried twice to unregister and it's still registered here we're obviously in a weird unexpected state, so trying again might just mask bigger issues.